File: app/src/main/java/com/eftimoff/bakingapp/recipedetails/view/RecipeIngredientsViewHolder.java
package com.eftimoff.bakingapp.recipedetails.view;

import android.view.View;
import android.widget.TextView;

import com.eftimoff.bakingapp.R;
import com.eftimoff.bakingapp.app.models.Ingredient;
import com.eftimoff.bakingapp.app.view.BaseViewHolder;

import java.util.List;

import butterknife.BindView;

public class RecipeIngredientsViewHolder extends BaseViewHolder<List<Ingredient>> {

    @BindView(R.id.recipeStepIngredientsList)
    TextView recipeStepIngredientsList;

    public RecipeIngredientsViewHolder(View itemView) {
        super(itemView);
    }

    @Override
    protected void bind(List<Ingredient> ingredients) {
        StringBuilder stringBuilder = new StringBuilder();

        for (int i = 0; i < ingredients.size(); i++) {
            Ingredient ingredient = ingredients.get(i);
            stringBuilder.append(ingredient.toString());
            if (i != ingredients.size() - 1) {
                stringBuilder.append("\n");
            }
        }
        recipeStepIngredientsList.setText(stringBuilder.toString());
    }

}

File: app/src/main/java/com/eftimoff/bakingapp/recipedetails/view/RecipeIngredientsViewHolder.kt
package com.eftimoff.bakingapp.recipedetails.view

import android.view.View
import com.eftimoff.bakingapp.app.extensions.formatQuantity
import com.eftimoff.bakingapp.app.models.Ingredient
import com.eftimoff.bakingapp.app.view.BaseViewHolder
import kotlinx.android.synthetic.main.item_recipe_step_ingredients.view.*

class RecipeIngredientsViewHolder(itemView: View) : BaseViewHolder<List<Ingredient>>(itemView) {

    override fun bind(t: List<Ingredient>) {
        val builder: StringBuilder = StringBuilder()
        t.forEachIndexed { index, it ->

            val quantity = it.quantity.formatQuantity(it.quantity)
            val measure = it.measure.toLowerCase()
            val ingredient = it.ingredient.toLowerCase()

            builder.append("$quantity $measure $ingredient")
            if (index != t.size - 1) {
                builder.append("\n")
            }

        }
        itemView.recipeStepIngredientsList.text = builder.toString()
    }

}